src/libs/actions/IOU.ts

```diff
@@ -4317,9 +4317,12 @@ function getUpdateMoneyRequestParams(
         (hasModifiedTag || hasModifiedCategory || hasModifiedComment || hasModifiedDistanceRate || hasModifiedAmount || hasModifiedCreated)
     ) {
         const currentTransactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`] ?? [];
+        const optimisticViolations = hasModifiedAmount
```

Comment:
Why only when `hasModifiedAmount` and not when currency changes or date changes too?